test/hotspot/gtest/runtime/test_signals.cpp

```diff
@@ -44,7 +46,7 @@ class PosixSignalTest : public ::testing::Test {
 
   static void check_handlers() {
     struct sigaction act, old_SIGFPE_act, old_SIGILL_act;
-    act.sa_handler = (void (*)(int))sig_handler;
+    act.sa_handler = CAST_FROM_FN_PTR(sa_handler_t, sig_handler);
```

Comment:
This is similarly wrong code that should be fixed instead of doing weird casting dances.  The problem here is that sig_handler has the wrong signature.  It should be just `void sig_handler(int)`.

Reply:
This one is even worse than the other.  Here we're installing a handler function taking 3 arguments in such a way that it will only be called with one.  Weee!

Reply:
:facepalm: